### ISSUE TITLE
Import functionality in Known

### ISSUE THREAD

**Issue by srcr** [idno/Known#480]
I would like to see a way you can import all posts in some form into Known.
In my opinion there is no need to make it feature rich, but a basic CVS import or similar would be great.

**Comment by michielbdejong** [comment]
@mindscratch see also https://github.com/idno/idno/issues/644 - may be an alternative if you're happy to work with the raw .sql files.

**Comment by benwerd** [comment]
As of right now, the export bundle includes the filesystem as Known stores it, as well as a full SQL entities export.

**Comment by michielbdejong** [comment]
Great!! When is this expected to be pushed to *.withknown.com instances?

**Comment by benwerd** [comment]
This week. I'll update this thread once the roll-out is complete.

**Comment by davidmead** [comment]
Does this mean I'll finally be able to get an export from `davidmead.withknown.com` and import it to `blog.davidjohnmead.com`?

**Comment by benwerd** [comment]
Yes. Sorry if we dropped the ball there. You'll be able to get it on demand.

**Comment by davidmead** [comment]
Great. Thanks Ben.

**Comment by socialreporter** [comment]
I have the same issue as davidmead - is it possible to export from mediablends.withknow.com (done that) and then import to my digital ocean site?

**Comment by michielbdejong** [comment]
@socialreporter Unfortunately it still requires a bit of fiddling, but I got it working in the end and [documented](https://github.com/idno/Known/issues/734#issuecomment-74413573) how I did it.

**Comment by michielbdejong** [comment]
@socialreporter I would be happy to help you via irc if/when you try this, I'm michielbdejong in #indiewebcamp on freenode (usually online most of European daytime).

**Comment by socialreporter** [comment]
Thanks Michiel

On 25 February 2015 at 14:16, Michiel de Jong <notifications@github.com>
wrote:

> @socialreporter <https://github.com/socialreporter> Unfortunately it
> still requires a bit of fiddling, but I got it working in the end and
> documented
> <https://github.com/idno/Known/issues/734#issuecomment-74413573> how I
> did it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/idno/Known/issues/480#issuecomment-75966197>.
>



-- 
+447970621696
http://socialreporter.com

**Comment by benwerd** [comment]
The export function on the hosted site and in the latest codebase now produces the metadata SQL.

**Comment by michielbdejong** [comment]
Oh, great!

**Comment by pierreozoux** [comment]
Can we close this thread now?

**Comment by davidmead** [comment]
OK. I still cannot import my Wordpress XML using the upload feature in the latest build.

Also, I exported from my hosted Known site, and now gives me a slew of new files I don't know what to with.

![davidmead_withknown_com_2015-08-29](https://cloud.githubusercontent.com/assets/1266930/9584714/b70184ec-4fe0-11e5-9533-2bb9d7cc9ee5.png)


Any help would be greatly appreciated.

**Comment by mapkyca** [comment]
Can close..?

**Comment by benwerd** [comment]
I briefly closed this, but no: import is not as functional as it needs to be, and unfortunately people are still reporting problems.

**Comment by mapkyca** [comment]
<a href="https://github.com/idno/Known/issues/480" class="u-in-reply-to"></a>Folk might want to kick the tires on the latest github head, I just pushed a bunch of tweaks which would be good to have a bunch of different output thrown at.

**Comment by davidmead** [comment]
I tired to import my WP export `.XML` file and it doesn't work.

This was using the 0.99 build from http://withknown.com - I'll update with 0.9.9a from you @mapkyca, but  do you know if the Wordpress imports works now?

**Comment by mapkyca** [comment]
<a href="https://github.com/idno/known/issues/480" class="u-in-reply-to"></a>To certain values of "work". Import / Export is a hard problem, with many variables/quirks and such and such that can cause issues, and it's not a well trodden path.

**Comment by lindner** [comment]
Would it be possible to add integration tests?

1) Import Test
- Add import file to repo.
- Run importer.
- Dump internal DB (mysqldump? something else?)
   - First time? save this as 'Golden' result.
- Subsequent runs verify that the data is imported correctly against the
golden DB

2) Export Test

- Perform test operations (create posts, etc.)
- Run Export, save Golden copy in repo.
- Subsequent runs compare with golden copy.
